src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php

```diff
@@ -148,29 +149,52 @@ protected function getEffectivePlugins(array $resources): array
      */
     protected function extractEffectivePlugins($effectivePlugins, $pluginsPerExporter): array
     {
-        foreach ($effectivePlugins as $effectivePlugin) {
-            if ($effectivePlugin instanceof EventResourceRepositoryPluginInterface || $effectivePlugin instanceof EventResourceBulkRepositoryPluginInterface) {
-                $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
+        foreach ($effectivePlugins as $effectiveEventPlugins) {
+            $effectivePlugin = $this->resolveDuplicatePlugins($effectiveEventPlugins);
+
+            if ($effectivePlugin === null) {
+                continue;
+            }
+
+            if ($this->isEventResourceRepositoryPlugin($effectivePlugin)) {
+                $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS][] = $effectivePlugin;
             }
 
             if ($effectivePlugin instanceof EventResourceQueryContainerPluginInterface) {
-                $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
+                $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS][] = $effectivePlugin;
             }
         }
 
-        $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS] = $this->resolveDuplicatePlugins($pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS]);
-        $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS] = $this->resolveDuplicatePlugins($pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS]);
-
         return $pluginsPerExporter;
     }
 
     /**
      * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[] $eventResourcePlugins
      *
-     * @return \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[]
+     * @return \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface|null
+     */
+    protected function resolveDuplicatePlugins(array $eventResourcePlugins): ?EventResourcePluginInterface
```

Comment:
This name is confusing, rename it to `findEffectivePlugin()`.

Reply:
Done.